app/repositories/onlinetesting/Phase3TestRepository.scala

```diff
@@ -86,7 +97,62 @@ class Phase3TestMongoRepository(dateTime: DateTimeFactory)(implicit mongo: () =>
     getTestGroup(applicationId, phaseName)
   }
 
-  override def getTestGroupByToken(token: String): Future[Phase3TestGroup] = {
-    getTestProfileByToken(token, phaseName)
+  override def getTestGroupByToken(token: String): Future[Phase3TestGroupWithAppId] = {
```

Comment:
We seriously need to look at pushing some of this stuff down into the base class. This level of code duplication is making me sad